demo-app/src/main/java/io/opentelemetry/android/demo/ui/shop/components/QuantityChooser.kt

```diff
@@ -0,0 +1,114 @@
+    quantity: Int,
+    onQuantityChange: (Int) -> Unit
+) {
+    var expanded by remember { mutableStateOf(false) }
+
+    Row(
+        verticalAlignment = Alignment.CenterVertically,
+        modifier = Modifier
+            .padding(vertical = 8.dp)
+            .fillMaxWidth()
+            .padding(horizontal = 16.dp)
+    ) {
+        Text(
+            text = "Quantity",
+            fontFamily = gothamFont,
+            fontSize = 18.sp,
+            modifier = Modifier.padding(end = 16.dp)
+        )
+        Box(
+            modifier = Modifier.weight(1f)
+        ) {
+            QuantityButton(quantity = quantity, onClick = { expanded = true })
+            QuantityDropdown(
+                expanded = expanded,
+                onDismissRequest = { expanded = false },
+                selectedQuantity = quantity,
+                onQuantitySelected = { selectedQuantity ->
+                    onQuantityChange(selectedQuantity)
+                    expanded = false
+                }
+            )
+        }
+    }
+}
+
+@Composable
+fun QuantityButton(quantity: Int, onClick: () -> Unit) {
+    OutlinedButton(
+        onClick = onClick,
+        modifier = Modifier
+            .width(120.dp)
+            .height(48.dp)
+    ) {
+        Row(
+            horizontalArrangement = Arrangement.SpaceBetween,
+            modifier = Modifier.fillMaxWidth()
+        ) {
+            Text(
+                text = quantity.toString(),
+                fontSize = 18.sp,
+                modifier = Modifier.align(Alignment.CenterVertically)
+            )
+            Icon(
+                imageVector = Icons.Filled.ArrowDropDown,
+                contentDescription = "Dropdown Icon",
+                modifier = Modifier.align(Alignment.CenterVertically)
+            )
+        }
+    }
+}
+
+@Composable
+fun QuantityDropdown(
+    expanded: Boolean,
+    onDismissRequest: () -> Unit,
+    selectedQuantity: Int,
+    onQuantitySelected: (Int) -> Unit
+) {
+    DropdownMenu(
+        expanded = expanded,
+        onDismissRequest = onDismissRequest
+    ) {
+        for (i in 1..10) {
+            DropdownMenuItem(
+                text = {
+                    Row(
+                        verticalAlignment = Alignment.CenterVertically
+                    ) {
+                        if (i == selectedQuantity) {
+                            Icon(
+                                imageVector = Icons.Filled.Check,
+                                contentDescription = "Selected",
+                                tint = Color.Black,
+                                modifier = Modifier.size(18.dp)
+                            )
+                            Spacer(modifier = Modifier.width(8.dp))
+                        }
+                        Text(
+                            text = i.toString(),
+                        )
+                    }
+                },
+                onClick = { onQuantitySelected(i) }
+            )
```

Comment:
I like that this was broken out into its own file. I think @LikeTheSalad had a comment previously about indent levels, and there's still some room to improve this. If you make a new topmost item called something like `QuantityDropDownItem` (whose constructor takes `i` and a boolean `isSelected`) then you can basically move this whole block into that new composable and save 1 or 2 indent levels. The for loop then looks something like
```
for (i in 1..10) {
    QuantityDropDownItem(i, i == selectedQuantity)
}
```